netlink.go

```diff
@@ -492,6 +484,37 @@ func assembleDestination(attrs []syscall.NetlinkRouteAttr) (*Destination, error)
 	return &d, nil
 }
 
+// getIPFamily parses the IP family based on raw data from netlink.
+// For AF_INET, netlink will set the first 4 bytes with trailing zeros
+//   10.0.0.1 -> [10 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0]
+// For AF_INET6, the full 16 byte array is used:
+//   2001:db8:3c4d:15::1a00 -> [32 1 13 184 60 77 0 21 0 0 0 0 0 0 26 0]
+func getIPFamily(address []byte) (uint16, error) {
```

Comment:
@aojea @uablrek can I get your review on this? Is this a safe way to determine the address family? It wouldn't account for addresses like `2600::` but that seems okay? 

Can't use Go's net package because it expects IPv4 address to be last 4 bytes but netlink returns it as first 4 bytes.

Reply:
sorry for my question but I'm not much familiar with this code, but assuming there is no nat64 or nat46, is not possible to obtain the IP family from the virtual IP or in another place and pass it as parameter, instead of  using the destination?
https://github.com/moby/ipvs/blob/e63ad1cd1c37ec36512a5c6963150b6da025cc30/netlink.go#L156

Reply:
scratch my comment, this https://github.com/moby/ipvs/pull/19#issuecomment-620526840 has a point :)

Reply:
Yeah worth clarifying that newer kernels pass the address family attribute so this code path would never run. But for old kernels we need some way to figure out the address family so we know how to parse the address into a net.IP.